src/ui/public/doc_table/components/table_row.js

```diff
@@ -77,6 +78,35 @@ module.directive('kbnTableRow', function ($compile) {
 
         $detailsScope.row = $scope.row;
 
+        $detailsScope.copyTextToClipboard = text => {
```

Comment:
I'd put it in its own folder under src/ui/public. Here's a simple example of such a service: https://github.com/elastic/kibana/blob/4c535f615500f78294726b4d087722bfa655c61e/src/ui/public/debounce/debounce.js#L8-L8